std/typecons.d

```diff
@@ -768,6 +770,12 @@ if (distinctFieldNames!(Specs))
             return field[] == rhs.field[];
         }
 
+        bool opEquals(R...)(R rhs) const
+        if (R.length > 1 && areCompatibleTuples!(typeof(this), Tuple!R, "=="))
+        {
+            return field[] == tuple(rhs).field[];
```

Comment:
This is wasteful if `field[] == rhs.field[]` works, which would be more efficient. Why is there a need to create a temp?

Reply:
Because `R` isn't a tuple - without it I get:

```
std/typecons.d(776): Error: no property field for tuple (string, string)
```

Reply:
Got it, thanks. Well the template constraint would at best describe what the requirements on R are; currently it relies on `areCompatibleTuples`, which is utterly undocumented.

Reply:
https://dlang.org/phobos/std_typecons.html#.Tuple.opEquals no?